src/Components/ProductCard/index.js

```diff
@@ -1,20 +1,71 @@
 /* eslint-disable jsx-a11y/anchor-is-valid */
+import { useState, useEffect } from 'react';
 import classnames from 'classnames';
 
 import moneyFormatter from '../../utilities/moneyFormatter';
 import './style.scss';
 
 const ProductCard = ({ product, className }) => {
+  const [selectedVariant, setSelectedVariant] = useState(false);
+  const [selectedColor, setSelectedColor] = useState(false)
+
   const onClick = () => {
     sessionStorage.productDetails = JSON.stringify(product);
     window.location = `/product-details?product=${product.productID}`;
   }
+
+  const colorMetaName = 'Color';
+  const maxColorOptionsToShow = 5;
+
+  const getColorOptions = () => {
+    const colorOptions = product?.meta_data.filter(md => md.key === colorMetaName);
+    return !!colorOptions && !!colorOptions[0] ? colorOptions[0].values : false;
+  }
+
+  const getSelectableVariantByColorOption = colorOptionValue => {
+    return product?.variants.find(variant => variant.meta_data.find(md => md.key === colorMetaName && md.value === colorOptionValue))
+  }
+
+  const selectColorOption = colorOptionValue => {
+    setSelectedColor(colorOptionValue);
+    if(!!colorOptionValue) {
+      const selectableVariant = getSelectableVariantByColorOption(colorOptionValue);
+      setSelectedVariant(selectableVariant)
+    }
+  }
+
+  useEffect(() => {
+    const colorOptions = getColorOptions();
+    selectColorOption(!!colorOptions ? colorOptions[0] : false)
+  }, [])
+
   return (
-    <div onClick={onClick} className={classnames("product-card d-flex flex-md-column", className)}>
-      <div className="product-card_img-wrapper overflow-hidden rounded-2">
-        <img src={product.images[0].src} alt="product" />
+    <div className={classnames("product-card d-flex flex-md-column align-items-center", className)}>
+      <div onClick={onClick} className="product-card_img-wrapper overflow-hidden rounded-2">
+        <img src={!!selectedVariant && !!selectedVariant?.image?.src ? selectedVariant?.image?.src : product.images[0].src} alt="product" />
       </div>
-      <div className="d-flex flex-column product-card_details ms-2">
+      {product.variants.length > 1 && getColorOptions() ?
+        <div className="product-card_variants">
+          {getColorOptions()/*.slice(0, maxColorOptionsToShow)*/.map((colorOption, i) => (
+            <div
+              className={selectedColor == colorOption ? "active" : ""}
```

Comment:
@hemant-cbotics can you change this to triple equal?

Reply:
@Nziranziza - I have updated this.